src/include/migraphx/value.hpp

```diff
@@ -188,6 +189,12 @@ struct value
     const cpp_type* if_##vt() const;
     MIGRAPHX_VISIT_VALUE_TYPES(MIGRAPHX_VALUE_GENERATE_DECL_METHODS)
 
+    template <class T>
+    using literal_to_string = std::conditional_t<(std::is_convertible<T, const char*>{} and
```

Comment:
It is returning either string or T. I can't think of better alternatives. Should it be named `literal_to_type` ? or `literal_to_T`  or `lit_to_str_T` ?

Reply:
If its a string literal its converted to std::string otherwise it just return T.